### PR TITLE
[Track 1-B] Add reflection_pad2d_backward operator

### PR DESCRIPTION
**Pull request by AdamPlatin123**
[FlagGems Operator Development Competition]

Add  operator to complete the backward pass for reflection_pad2d.

## Implementation
- Triton JIT kernel with atomic_add for gradient accumulation
- Follows same pattern as 
- Float32 intermediate accumulation for precision
- Handles 2D padding with reflection logic

## Changes
- : New operator
- : Accuracy tests
- : Performance benchmark
- : Register operator

### CODE REVIEW

**Comment by bin913** [comment]
@AdamPlatin123 Please fix failing check

File: src/flag_gems/ops/reflection_pad2d_backward.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    # Initialize grad_input as float32 for safe accumulation
+    grad_input = torch.zeros_like(x, dtype=torch.float32)
+
+    # No padding case - just copy
+    if pad_left == 0 and pad_right == 0 and pad_top == 0 and pad_bottom == 0:
+        if H_out != H_in or W_out != W_in:
+            raise RuntimeError(
+                "Internal error: H_out/W_out should equal H_in/W_in when no padding"
+            )
+        BLOCK_HW = 256
```
Comment (bin913) [inline src/flag_gems/ops/reflection_pad2d_backward.py:99]: **Suggestion**: Hardcoded hyperparameter without comment: BLOCK_HW = 256

**Recommended fix:**
Add a comment explaining why this value was chosen.

<sub>Rule: `HYPERPARAMS`</sub>

File: src/flag_gems/ops/reflection_pad2d_backward.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    if H_in < 2 or W_in < 2:
+        raise ValueError(
+            "input spatial dimensions must be at least 2 for reflection padding when padding > 0"
+        )
+    if pad_left >= W_in or pad_right >= W_in or pad_top >= H_in or pad_bottom >= H_in:
+        raise ValueError(
+            "padding values must be less than the input spatial dimensions for reflection padding"
+        )
+
+    BLOCK_HW = 256
```
Comment (bin913) [inline src/flag_gems/ops/reflection_pad2d_backward.py:124]: **Suggestion**: Hardcoded hyperparameter without comment: BLOCK_HW = 256

**Recommended fix:**
Add a comment explaining why this value was chosen.

<sub>Rule: `HYPERPARAMS`</sub>

**Review by bin913 — CHANGES_REQUESTED** [review]
## FlagGems Automated Review

**Operator:** `reflection_pad2d_backward`
**Verdict:** REQUEST_CHANGES

### Summary
- Errors: 1
- Warnings: 2
- Suggestions: 0

### Issues (no specific line)
- **[ERROR]** `REG_CONSISTENCY`: PR does not include changes to conf/operators.yaml or src/flag_gems/__init__.py. The operator may not be fully registered. — *Fix: Add yaml entry in conf/operators.yaml and _FULL_CONFIG entry in src/flag_gems/__init__.py for reflection_pad2d_backward.*

File: src/flag_gems/ops/reflection_pad2d_backward.py (hunk 1 above)
Comment (bin913) [inline src/flag_gems/ops/reflection_pad2d_backward.py:99]: **Suggestion**: Hardcoded hyperparameter without comment: BLOCK_HW = 256

**Recommended fix:**
Add a comment explaining why this value was chosen.

<sub>Rule: `HYPERPARAMS`</sub>

File: src/flag_gems/ops/reflection_pad2d_backward.py (hunk 2 above)
Comment (bin913) [inline src/flag_gems/ops/reflection_pad2d_backward.py:124]: **Suggestion**: Hardcoded hyperparameter without comment: BLOCK_HW = 256

**Recommended fix:**
Add a comment explaining why this value was chosen.

<sub>Rule: `HYPERPARAMS`</sub>

**Review by bin913 — CHANGES_REQUESTED** [review]
## FlagGems Automated Review

**Operator:** `reflection_pad2d_backward`
**Verdict:** REQUEST_CHANGES

### Summary
- Errors: 1
- Warnings: 2
- Suggestions: 0

### Issues (no specific line)
- **[ERROR]** `REG_CONSISTENCY`: PR does not include changes to conf/operators.yaml or src/flag_gems/__init__.py. The operator may not be fully registered. — *Fix: Add yaml entry in conf/operators.yaml and _FULL_CONFIG entry in src/flag_gems/__init__.py for reflection_pad2d_backward.*

**Comment by AdamPlatin123** [comment]
@bin913 Thank you for the review!

✅ Added `reflection_pad2d_backward` to `conf/operators.yaml`
✅ Rebased onto latest master
✅ Fixed code-style (isort + black formatting)
✅ Squashed into single clean commit

Please re-check when CI reruns.

**Comment by bin913** [comment]
@AdamPlatin123 Please fix failing checks
